test/datasets_utils.py

```diff
@@ -273,7 +321,8 @@ def inject_fake_data(self, tmpdir: str, config: Dict[str, Any]) -> Union[int, Di
         Args:
             tmpdir (str): Path to a temporary directory. For most cases this acts as root directory for the dataset
                 to be created and in turn also for the fake data injected here.
-            config (Dict[str, Any]): Configuration that will be used to create the dataset.
+            config (Dict[str, Any]): Configuration that will be used to create the dataset. It provides at least fields
```

Comment:
Maybe a slightly more explicit description:

```suggestion
            config (Dict[str, Any]): Configuration that will be passed to the dataset constructor. It provides at least fields
```

Also:

> It provides at least fields for all dataset parameters with default values

Should we prefix that by something like "When ``inject_fake_data()`` is called by ``create_dataset()``, ``config`` provides at least..."

Reply:
Given that I can't imagine a case where one would call `inject_fake_data` directly, I would leave this for now.